okhttp/src/main/java/okhttp3/internal/ws/RealWebSocket.kt

```diff
@@ -144,6 +144,12 @@ class RealWebSocket(
   }
 
   fun connect(client: OkHttpClient) {
+    if (originalRequest.header("Sec-WebSocket-Extensions") != null) {
+      failWebSocket(ProtocolException(
+          "Request header not permitted: 'Sec-WebSocket-Extensions'"), null)
+      return
```

Comment:
@swankjesse I think this code is a little too restrictive to applications that want to specify their own extensions or other extensions that are not just the `permessage-deflate`.
I would rather merge the extensions provided by the application with the internal `permessage-deflate`, unless the application already specifies `permessage-deflate`. Thoughts?
FTR I have a CometD test that was testing this that fails with OkHttp 4.5.0. Thanks!

Reply:
Can you raise a separate issue to track and link to your test case?  Want to make sure this doesn't get lost if we don't immediately fix.

Reply:
@yschimke filed #5953, thanks!